### PR TITLE
VizRepeater/BarGauge: Use common font dimensions across repeated visualisations

### PR DESCRIPTION
**Pull request by ryantxu**
Currently each vis within a VizRepeater calculates its own font size etc.  This can look weird when they do not use the same values.

This PR adds scaffolding to calculate the value once and pass it to each vis.

TODO:

- [ ] Agree on property/function naming
- [ ] Apply this to BarGauge
- [ ] Apply this to Gauge

### CODE REVIEW

File: packages/grafana-ui/src/components/VizRepeater/VizRepeater.tsx [hunk 1]
```diff
@@ -1,12 +1,23 @@
...
-interface Props<T> {
-  renderValue: (value: T, width: number, height: number) => JSX.Element;
+interface Props<V, D> {
+  /**
+   * Optionally precalculate dimensions to support consistent behavior between repeated
+   * values.  Two typical patterns are:
+   * 1) Calculate raw values like font size etc and pass them to each vis
+   * 2) find the maximum input values and pass that to the vis
+   */
+  calculateInternalDimensions?: (values: V[], width: number, height: number) => D;
```
Comment (torkelo) [inline packages/grafana-ui/src/components/VizRepeater/VizRepeater.tsx]: getAlignmentOptions or 
getAlignmentFactors

**Review by torkelo — COMMENTED** [review]
Was worried that you would continue working on this on the plane so just added some comments on naming for now. But let me know if I can take over this.

**Comment by ryantxu** [comment]
@torkelo -- I am not actively working on this, please take it over if it is helpful

**Comment by torkelo** [comment]
Will do!

**Review by torkelo — APPROVED** [review]
LGTM now :)

**Pull request merged**